### PR TITLE
ADD basic playlist support + youtube playlist support

### PR DESCRIPTION
**Pull request by BlenderViking**
Hi,

I begin the refactor of my previous pull request : https://github.com/TeamNewPipe/NewPipe/pull/377

To avoid the same problem as previously encountered (to big code base change), I begin to submit multiple small pull request.

To avoid the same problem as previously encountered, I will send the code by multiple small pull request.

So here the code for import an external playlist in the core extractor app.

I made a unit test but I didn't found how to commit it in the core part.
So i made a gist for store it : https://gist.github.com/BlenderViking/c73c6776580ed661f78f6b0739a66e73

### CODE REVIEW

**Comment by theScrabi** [comment]
The unit tests are still part of the NewPipe app itself.

Nice work. :)

File: StreamingService.java [hunk 1]
```diff
@@ -52,8 +52,11 @@ public abstract StreamExtractor getExtractorInstance(String url)
     public abstract SearchEngine getSearchEngineInstance();
     public abstract UrlIdHandler getStreamUrlIdHandlerInstance();
     public abstract UrlIdHandler getChannelUrlIdHandlerInstance();
+	public abstract UrlIdHandler getPlayListUrlIdHandlerInstance();
```
Comment (theScrabi) [inline StreamingService.java]: Please fix the spacing.

**Comment by BlenderViking** [comment]
Ok, It's fixed.
I just a question, how to correctly map the git submodules on Android Studio in the main project ?

**Comment by theScrabi** [comment]
Ow good question, I did it using the command line.

**Comment by theScrabi** [comment]
I've [updated](https://github.com/TeamNewPipe/NewPipe/commit/8735cf931adcd2a3c711a7ccbcb386e1696bd5aa) the extractor repository.

**Comment by BlenderViking** [comment]
I found, after clone the repos, we just need to tape in command line :
The init the submodule :
- git submodule init
The upgrade of the submodule :
- git submodule update --recursive --remote
And then in AndroidStudio define in the git config the submodule : 
submodule integration
And we can see the repos of the submodule in Android Studio : 
image

**Comment by theScrabi** [comment]
ah alright

**Pull request merged**